Xamarin.Forms.Platform.iOS/ContextScrollViewDelegate.cs

```diff
@@ -140,50 +147,57 @@ public override void WillEndDragging(UIScrollView scrollView, PointF velocity, r
 			{
 				IsOpen = true;
 				targetContentOffset = new PointF(width, 0);
-				RemoveHighlight(scrollView);
+				RemoveHighlight();
 
-				if (_globalCloser == null)
-				{
-					UIView view = scrollView;
-					while (view.Superview != null)
-					{
-						view = view.Superview;
-
-						NSAction close = () =>
-						{
-							if (UIDevice.CurrentDevice.CheckSystemVersion(8, 0))
-								RestoreHighlight(scrollView);
-
-							IsOpen = false;
-							scrollView.SetContentOffset(new PointF(0, 0), true);
-
-							ClearCloserRecognizer(scrollView);
-						};
-
-						var table = view as UITableView;
-						if (table != null)
-						{
-							_table = table;
-							_globalCloser = new GlobalCloseContextGestureRecognizer(scrollView, close);
-							_globalCloser.ShouldRecognizeSimultaneously = (recognizer, r) => r == _table.PanGestureRecognizer;
-							table.AddGestureRecognizer(_globalCloser);
-
-							_closer = new UITapGestureRecognizer(close);
-							var cell = GetContextCell(scrollView);
-							cell.ContentCell.AddGestureRecognizer(_closer);
-						}
-					}
-				}
+				AddClosers();
 			}
 			else
 			{
-				ClearCloserRecognizer(scrollView);
+				ClearCloserRecognizer();
 
 				IsOpen = false;
 				targetContentOffset = new PointF(0, 0);
 
 				if (UIDevice.CurrentDevice.CheckSystemVersion(8, 0))
-					RestoreHighlight(scrollView);
+					RestoreHighlight();
+			}
+		}
+
+		private void AddClosers()
+		{
+			NSAction close = () =>
+			{
+				if (UIDevice.CurrentDevice.CheckSystemVersion(8, 0))
+					RestoreHighlight();
+
+				IsOpen = false;
+				_scrollView.SetContentOffset(new PointF(0, 0), true);
+
+				ClearCloserRecognizer();
+			};
```

Comment:
Move `close` declaration outside the `while` loop to avoid unnecessary initialization in each iteration.